SportsHub/SportsHub.DAL/Repository/GenericRepository.cs

```diff
@@ -0,0 +1,36 @@
+﻿using SportsHub.DAL.Data;
+using SportsHub.Domain.Repository;
+
+namespace SportsHub.DAL.Repository
+{
+    public class GenericRepository<T> : IGenericRepository<T>
+        where T : class
+    {
+        protected readonly ApplicationDbContext context;
+
+        public GenericRepository(ApplicationDbContext context)
+        {
+            this.context = context;
+        }
+
+        public void Add(T entity)
+        {
+            context.Set<T>().Add(entity);
+        }
+
+        public void Delete(T entity)
+        {
+            context.Set<T>().Remove(entity);
```

Comment:
lets avoid context.Set<T> everywhere. Add DbSet<T> DbSet => Context.Set<T>(). and use DbSet everywhere

Reply:
Implemented